extensions/parquet/table/src/main/java/io/deephaven/parquet/table/pagestore/topage/ToBigDecimalFromNumeric.java

```diff
@@ -0,0 +1,70 @@
+//
+// Copyright (c) 2016-2024 Deephaven Data Labs and Patent Pending
+//
+package io.deephaven.parquet.table.pagestore.topage;
+
+import io.deephaven.chunk.ChunkType;
+import io.deephaven.chunk.attributes.Any;
+import io.deephaven.parquet.base.PageMaterializerFactory;
+import io.deephaven.parquet.base.materializers.BigDecimalFromIntMaterializer;
+import io.deephaven.parquet.base.materializers.BigDecimalFromLongMaterializer;
+import org.jetbrains.annotations.NotNull;
+
+import java.math.BigDecimal;
+
+public class ToBigDecimalFromNumeric<ATTR extends Any> implements ToPage<ATTR, BigDecimal[]> {
+
+    public static <ATTR extends Any> ToPage<ATTR, BigDecimal[]> createFromInt(
```

Comment:
You suggested you could just take the primitive type and simplify this code. Probably `ToLongPage`, too. I'm not sure that's better, since it moves the logic down here, rather than the code where we determine the types we're handling. That said, the `FromInt` and `FromLong` classes are adding no value and should be removed from the hierarchy.